Zend/tests/attributes/001_placement.phpt

```diff
@@ -0,0 +1,112 @@
+--TEST--
+Attributes can be placed on all supported elements.
+--FILE--
+<?php
+
+<<A1(1)>>
+class Foo
+{
+    <<A1(2)>>
+    public const FOO = 'foo', BAR = 'bar';
+    
+    <<A1(3)>>
+    public $x, $y;
```

Comment:
Hm ... this behavior is questionable, and does not seem to be mentioned in the RFC. At least this does not match how PHP interprets doc comments, which only apply to the directly following property:

```
<?php

class Test {
    /** doc 1 */
    public $x, $y;
     
    public 
        /** doc 2 */
        $a, 
        /** doc 3 */ 
        $b;
}   
 
$rc = new ReflectionClass(Test::class); 
var_dump($rc->getProperty('y')->getDocComment()); // bool(false)
var_dump($rc->getProperty('b')->getDocComment()); // doc 3
```

Reply:
This has not been addressed yet. What shall we do about it? Apply the attribute to all properties or only to the first one? The current implementation treats attributtes like additional visibility modifiers and applies them to all properties. The same thing has to be done for class constants.

Reply:
Since attributes behave the same as doc comments for inheritance, we should also use the same behavior here. For consistency reasons users probably expect them to work this way. To me doc comments are user readable metadata, attributes machine readable metadata. So their behavior should be the same.

Reply:
I'm not sure on this one. In some sense aligning with doc comments makes sense, but doc comments also have different syntactic requirements. While you can write

```
    public 
        /** doc 2 */
        $a, 
        /** doc 3 */ 
        $b;
```

you cannot write

```
    public 
        <<Attribute2>>
        $a, 
        <<Attribute3>>
        $b;
```

As the syntax is limited to the whole property declaration group, rather than individual properties, it probably does make sense to apply it to all properties in the group, even if it's not consistent with doc comments.

Reply:
Uff, would documenting that attributes never apply to second+ properties in the multi statement be enough? This feels like a weird edge case.

Reply:
Another possibility is to make this an error. No attributes allowed if multiple properties declared in one group.

Reply:
Ah yes, that is better.